File: lab/javascript/segmentos/segmentos.js
function contar_segmentos(numeros){
    if (numeros.length == 0){
        return 0;
    }

    let segmentos = 1;
    let anterior = numeros[0];

    for (let i=1; i < numeros.length; i++){
        if (numeros[i] != anterior){
            segmentos ++;
            anterior = numeros[i];
        }
    }

    return segmentos;
}


console.log(contar_segmentos([5, 2, 2, 3, 4, 4, 4, 4, 1, 1]));
console.log(contar_segmentos([2, 2, 2, 3, 4, 4, 4, 4, 1, 1]));

File: lab/python/segmentos/segmentos.py
def contar_segmentos(*numeros):
    if not numeros:
        return 0

    segmentos = 1  # o primeiro número já inicia o primeiro segmento
    anterior = numeros[0]

    for numero in numeros[1:]:
        if numero != anterior:
            segmentos += 1
            anterior = numero

    return segmentos


print(contar_segmentos(5, 2, 2, 3, 4, 4, 4, 4, 1, 1))  # saída esperada: 5
print(contar_segmentos(2, 2, 2, 3, 4, 4, 4, 4, 1, 1))  # saída esperada: 5
